dvc/stage/__init__.py

```diff
@@ -482,15 +483,40 @@ def save_deps(self, allow_missing=False):
                 if not allow_missing:
                     raise
 
-    def save_outs(self, allow_missing=False):
+    def save_outs(
+        self, allow_missing: bool = False, merge_versioned: bool = False
+    ):
         from dvc.output import OutputDoesNotExistError
 
+        from .exceptions import StageFileDoesNotExistError, StageNotFound
+
+        if merge_versioned:
+            try:
+                old = self.reload()
```

Comment:
This extra `reload` on save might be expensive, so it's tied to the `merge_versioned` flag for now.

The problem is that we can't do this in the same place we are doing `restore_fields`. `restore_fields` is done on `repo.stage.create` and is intended to set fields that may be replaced/updated afterwards (i.e. we load original annotations from a dvcfile in `restore_fields` and then use `out.annot.update()` later to replace the loaded values with any new/updated ones from the command line.

But for the version metadata merge, we cannot do it until after the new outs have been `save()`'d, since we need the new out md5's to be computed before we can compare them to the old ones.